test/lib/resources/kube.go

```diff
@@ -54,19 +54,19 @@ func eventSenderPodImage(imageName string, name string, sink string, event *clou
 	if event.Encoding == "" {
 		event.Encoding = cloudevents.DefaultEncoding
 	}
-	eventExtensionsBytes, err := json.Marshal(event.Extensions)
+	eventExtensionsBytes, err := json.Marshal(event.Event.Extensions())
 	eventExtensions := string(eventExtensionsBytes)
 	if err != nil {
 		return nil, fmt.Errorf("encountered error when we marshall cloud event extensions %v", err)
 	}
 
 	args := []string{
 		"-event-id",
-		event.ID,
+		event.Event.ID(),
```

Comment:
More i look at that, more i think that this event sender image should just get the event encoded as json as input argument, it will allow to remove all that code here to encode the event into the program arguments

Reply:
I am not reworking the code here, I am just updating to sdk v2